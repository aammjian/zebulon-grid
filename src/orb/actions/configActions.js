import { Config } from '../Config';
import { AxisType } from '../Axis';

export const SET_CONFIG = 'SET_CONFIG';
export const setConfig = configObject => ({
  type: SET_CONFIG,
  config: new Config(configObject),
});

export const TOGGLE_DATAFIELD = 'TOGGLE_DATAFIELD';
export const toggleDatafield = datafieldId => ({
  type: TOGGLE_DATAFIELD,
  id: datafieldId,
});

export const ADD_FIELD = 'ADD_FIELD';
export const addField = (fieldId, axis, position) => ({
  type: ADD_FIELD,
  id: fieldId,
  axis: axis === AxisType.ROWS ? 'rows' : 'columns',
  position,
});

export const REMOVE_FIELD = 'REMOVE_FIELD';
export const removeField = (fieldId, axis) => ({
  type: REMOVE_FIELD,
  id: fieldId,
  axis: axis === AxisType.ROWS ? 'rows' : 'columns',
});
