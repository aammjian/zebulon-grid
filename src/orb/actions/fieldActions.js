export const CHANGE_SORT_ORDER = 'CHANGE_SORT_ORDER';
export const changeSortOrder = fieldId => ({
  type: CHANGE_SORT_ORDER,
  field: fieldId,
});
