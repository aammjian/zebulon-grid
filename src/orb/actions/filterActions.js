import { ExpressionFilter } from '../Filtering';

export const ADD_FILTER = 'ADD_FILTER';
export const addFilter =
  (fieldId, operator, term, staticValue, excludeStatic) => ({
    type: ADD_FILTER,
    filter: new ExpressionFilter(fieldId, operator, term, staticValue, excludeStatic),
    field: fieldId,
  });

export const DELETE_FILTER = 'DELETE_FILTER';
export const deleteFilter = fieldId => ({
  type: DELETE_FILTER,
  field: fieldId,
});
