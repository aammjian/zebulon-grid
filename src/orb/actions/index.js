export * from './dataActions';
export * from './configActions';
