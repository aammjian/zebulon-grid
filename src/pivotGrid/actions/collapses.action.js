import { AxisType, toAxis } from '../Axis';
import { EXPAND_COLLAPSE } from '../constants';

/* eslint-disable import/prefer-default-export */
export const toggleCollapse = ({ axisType, key }) => {
	return {
		type: EXPAND_COLLAPSE,
		key,
		axis: toAxis(axisType)
	};
};
/* eslint-enable */
