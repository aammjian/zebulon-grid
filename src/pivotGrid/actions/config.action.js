import { fieldFactory, datafieldFactory } from '../fields';
import {
  SET_FIELDS,
  SET_DATAFIELDS,
  SET_CONFIG_PROPERTY,
  TOGGLE_DATAFIELD,
  MOVE_FIELD,
  ZOOM_IN,
  ZOOM_OUT
} from '../constants';

export const setFields = configObject => ({
  type: SET_FIELDS,
  fields: configObject.fields.map(field => fieldFactory(field))
});

export const setDatafields = configObject => ({
  type: SET_DATAFIELDS,
  datafields: configObject.datafields.map(field => datafieldFactory(field))
});

export const setConfigProperty = (configObject, property, defaultValue) => ({
  type: SET_CONFIG_PROPERTY,
  property,
  value: configObject[property] || defaultValue
});

export const toggleDatafield = datafieldId => ({
  type: TOGGLE_DATAFIELD,
  id: datafieldId
});

export const moveField = (fieldId, oldAxis, newAxis, position) => ({
  type: MOVE_FIELD,
  id: fieldId,
  oldAxis,
  newAxis,
  position
});

export const zoomIn = () => ({ type: ZOOM_IN });
export const zoomOut = () => ({ type: ZOOM_OUT });
