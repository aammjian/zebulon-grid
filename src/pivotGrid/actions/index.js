export * from './data.action';
export * from './config.action';
export * from './sizes.action';
export * from './dimension.action';
export * from './filter.action';
export * from './collapses.action';
export * from './selection.action';
