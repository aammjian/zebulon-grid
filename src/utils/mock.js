/* eslint-disable import/no-extraneous-dependencies*/
import { Observable } from "rx-lite";
/* eslint-enable */
// import * as 'from' './Format';

export function getMockDatasource(dataRepetition = 1, nToto = 10, nTiti = 10) {
  const nTutu = 2;
  let obj = [];
  const res = [];
  for (let k = 0; k < dataRepetition; k += 1) {
    for (let o = 0; o < nToto; o += 1) {
      for (let i = 0; i < nTiti; i += 1) {
        for (let u = 0; u < nTutu; u += 1) {
          obj = {};
          obj.toto = o;
          obj.toto_lb = `toto ${String(nToto - o)}`;
          obj.toto_0 = `att0 ${String(o)}`;
          obj.toto_1 = `att1 ${String(nToto - o)}`;
          obj.titi = `titi ${String(i)}`;
          obj.tutu = String(u);
          obj.qty = u + 10 * i + 100 * o + 1; // +9999999999.1234567890123456
          obj.amt = u + 10 * i + 100 * o + 1000; // +9999999999.1234567890123456
          res.push(obj);
        }
      }
    }
  }
  return res;
}
export function getMockDatasource2(dataRepetition = 1, nToto = 10, nTiti = 10) {
  const nTutu = 2;
  let obj = [];
  const res = [];
  for (let k = 0; k < dataRepetition; k += 1) {
    for (let o = 0; o < nToto; o += 1) {
      for (let i = 0; i < nTiti; i += 1) {
        for (let u = 0; u < nTutu; u += 1) {
          obj = {};
          obj.toto2 = o;
          obj.toto_lb = `toto2 ${String(nToto - o)}`;
          obj.toto_0 = `att0 ${String(o)}`;
          obj.toto_1 = `att1 ${String(nToto - o)}`;
          obj.titi = `titi2 ${String(i)}`;
          obj.tutu = String(u);
          obj.qty = u + 10 * i + 100 * o + 1; // +9999999999.1234567890123456
          obj.amt = u + 10 * i + 100 * o + 1000; // +9999999999.1234567890123456
          res.push(obj);
        }
      }
    }
  }
  return res;
}
export function getObservableMockDatasource(interval) {
  const data = [
    getMockDatasource(),
    [
      {
        toto: "0",
        toto_lb: "toto 0",
        qty: 100,
        amt: 100,
        titi: "titi 0",
        tutu: "1"
      },
      {
        toto: "0",
        toto_lb: "toto 0",
        qty: 100,
        amt: 100,
        titi: "titi 0",
        tutu: "0"
      }
    ],
    { toto: "0", toto_lb: "toto 0", qty: 1, amt: 2, titi: "titi 0", tutu: "1" }
  ];
  return Observable.interval(interval || 100)
    .take(3)
    .map(i => data[i]);
}
export function getObservableError() {
  return Observable.throw(new Error("titi"));
}
export const getPromiseMockDatasource = (
  dataRepetition = 1,
  nToto = 10,
  nTiti = 10
) => {
  const p = new Promise(resolve => setTimeout(resolve, 20)).then(
    () => getMockDatasource(dataRepetition, nToto, nTiti)
    // {
    //   throw new Error('toto');
    // }
  );
  return p;
};
export const basicConfig = {
  measureHeadersAxis: "columns",
  width: 1099,
  height: 601,
  cellHeight: 30,
  cellWidth: 100,

  dimensions: [
    {
      id: "toto",
      caption: "Toto",
      keyAccessor: "toto",
      labelAccessor: "toto_lb",
      sort: {
        keyAccessor: "toto_lb"
      }
    },
    {
      id: "titi",
      caption: "Titi",
      keyAccessor: "titi"
    },
    {
      id: "tutu",
      caption: "Tutu",
      keyAccessor: "tutu",
      format: "id"
    },
    {
      id: "toto att 0",
      caption: "toto 0",
      keyAccessor: "toto_0",
      attributeParents: ["toto"]
    },
    {
      id: "toto att 1",
      caption: "toto 1",
      keyAccessor: "toto_1",
      attributeParents: ["toto"],
      format: "id"
    }
  ],
  measures: [
    {
      valueAccessor: "qty",
      id: "qty",
      caption: "Quantity",
      format: "quantity",
      aggregation: "sum"
    },
    {
      valueAccessor: "amt",
      id: "amt",
      caption: "Amount",
      aggregation: "sum",
      aggregationCaption: "whatever",
      format: "amount"
    },
    {
      id: "price",
      caption: "Price",
      aggregation: "weighted_avg",
      valueAccessor: "price",
      format: "price"
    }
  ],
  columns: [],
  rows: ["toto", "toto att 0", "toto att 1", "tutu", "titi"],
  activeMeasures: ["qty", "amt", "price"],
  collapses: { rows: { 99: true, 98: true }, columns: {} }
};
export const basicConfig2 = {
  measureHeadersAxis: "columns",
  width: 1099,
  height: 601,
  cellHeight: 30,
  cellWidth: 100,

  dimensions: [
    {
      id: "toto2",
      caption: "Toto",
      keyAccessor: "toto2",
      labelAccessor: "toto_lb",
      sort: {
        keyAccessor: "toto_lb"
      }
    },
    {
      id: "titi",
      caption: "Titi",
      keyAccessor: "titi"
    },
    {
      id: "tutu",
      caption: "Tutu",
      keyAccessor: "tutu",
      format: "id"
    },
    {
      id: "toto att 0",
      caption: "toto 0",
      keyAccessor: "toto_0",
      attributeParents: ["toto2"]
    },
    {
      id: "toto att 1",
      caption: "toto 1",
      keyAccessor: "toto_1",
      attributeParents: ["toto2"],
      format: "id"
    }
  ],
  measures: [
    {
      valueAccessor: "qty",
      id: "qty",
      caption: "Quantity",
      format: "quantity",
      aggregation: "sum"
    },
    {
      valueAccessor: "amt",
      id: "amt",
      caption: "Amount",
      aggregation: "sum",
      aggregationCaption: "whatever",
      format: "amount"
    },
    {
      id: "price",
      caption: "Price",
      aggregation: "weighted_avg",
      valueAccessor: "price",
      format: "price"
    }
  ],
  columns: ["titi"],
  rows: ["toto2", "toto att 0", "toto att 1", "tutu"],
  activeMeasures: ["qty", "amt", "price"]
};
