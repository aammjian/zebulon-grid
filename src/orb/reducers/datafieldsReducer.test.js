import datafields from './datafieldsReducer';
import { basicConfig } from '../../utils/mock';
import { toggleDatafield, setConfig } from '../actions';

const state = datafields({}, setConfig(basicConfig));

describe('datafields', () => {
  it('can be toggled', () => {
    expect(datafields(state, toggleDatafield('qty'))).toMatchSnapshot();
  });
});
