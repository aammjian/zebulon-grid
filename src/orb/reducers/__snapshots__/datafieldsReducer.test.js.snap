exports[`datafields can be toggled 1`] = `
Object {
  "amt": Field {
    "_aggregatefunc": [Function],
    "_formatfunc": [Function],
    "aggregateFuncName": "whatever",
    "caption": "Amount",
    "code": "amt",
    "columnSettings": Field {
      "_formatfunc": [Function],
      "aggregateFuncName": null,
      "caption": undefined,
      "code": undefined,
      "name": "amt",
      "sort": SortConfig {
        "customfunc": undefined,
        "order": null,
      },
      "subTotal": SubTotalConfig {
        "collapsed": undefined,
        "collapsible": undefined,
        "visible": undefined,
      },
    },
    "dataSettings": Field {
      "_formatfunc": [Function],
      "aggregateFuncName": null,
      "caption": undefined,
      "code": undefined,
      "name": "amt",
      "sort": SortConfig {
        "customfunc": undefined,
        "order": null,
      },
      "subTotal": SubTotalConfig {
        "collapsed": undefined,
        "collapsible": undefined,
        "visible": undefined,
      },
    },
    "name": "amt",
    "rowSettings": Field {
      "_formatfunc": [Function],
      "aggregateFuncName": null,
      "caption": undefined,
      "code": undefined,
      "name": "amt",
      "sort": SortConfig {
        "customfunc": undefined,
        "order": null,
      },
      "subTotal": SubTotalConfig {
        "collapsed": undefined,
        "collapsible": undefined,
        "visible": undefined,
      },
    },
    "sort": SortConfig {
      "customfunc": undefined,
      "order": null,
    },
    "subTotal": SubTotalConfig {
      "collapsed": undefined,
      "collapsible": undefined,
      "visible": undefined,
    },
  },
  "qty": Object {
    "_aggregatefunc": [Function],
    "_formatfunc": [Function],
    "activated": false,
    "aggregateFuncName": "sum",
    "caption": "Quantity",
    "code": "qty",
    "columnSettings": Field {
      "_formatfunc": [Function],
      "aggregateFuncName": null,
      "caption": undefined,
      "code": undefined,
      "name": "qty",
      "sort": SortConfig {
        "customfunc": undefined,
        "order": null,
      },
      "subTotal": SubTotalConfig {
        "collapsed": undefined,
        "collapsible": undefined,
        "visible": undefined,
      },
    },
    "dataSettings": Field {
      "_formatfunc": [Function],
      "aggregateFuncName": null,
      "caption": undefined,
      "code": undefined,
      "name": "qty",
      "sort": SortConfig {
        "customfunc": undefined,
        "order": null,
      },
      "subTotal": SubTotalConfig {
        "collapsed": undefined,
        "collapsible": undefined,
        "visible": undefined,
      },
    },
    "name": "qty",
    "rowSettings": Field {
      "_formatfunc": [Function],
      "aggregateFuncName": null,
      "caption": undefined,
      "code": undefined,
      "name": "qty",
      "sort": SortConfig {
        "customfunc": undefined,
        "order": null,
      },
      "subTotal": SubTotalConfig {
        "collapsed": undefined,
        "collapsible": undefined,
        "visible": undefined,
      },
    },
    "sort": SortConfig {
      "customfunc": undefined,
      "order": null,
    },
    "subTotal": SubTotalConfig {
      "collapsed": undefined,
      "collapsible": undefined,
      "visible": undefined,
    },
  },
}
`;
